Use CheckedPtr for complex lifetime objects in RenderBox.cpp

kate-k-lee opened the pull request:
#### 98bcd593fdaa08079c746e6f1f07e09ac3365f01
<pre>
Use CheckedPtr for complex lifetime objects in RenderBox.cpp
<a href="https://bugs.webkit.org/show_bug.cgi?id=297189">https://bugs.webkit.org/show_bug.cgi?id=297189</a>

Reviewed by Lily Spiniolas.

This patch updates auto* declarations for objects with complex lifetimes (such as ScrollableArea) to CheckedPtr to improve debugging safety while maintaining rendering performance.

Raw pointers are kept unchanged for DOM/tree navigation results where lifetime is well-defined and performance is critical.

No functional behavior changes are introduced.

Canonical link: <a href="https://commits.webkit.org/298943@main">https://commits.webkit.org/298943@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/e87f90ba682184aa106042a8e798c0f772db9f2e

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/116963 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/36628 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/27204 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/123048 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/68981 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/118852 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/37325 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/45228 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/88833 "Passed tests") | [❌ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/43551 "") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/119912 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/29771 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/104931 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/69294 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/28839 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/23039 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/66704 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/99161 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/23192 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/126175 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/43863 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/32972 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/97495 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/44228 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/101133 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/97297 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/24814 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/42632 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/20569 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/40258 "The change is no longer eligible for processing.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/43748 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/49344 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/43215 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/46554 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/44920 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

lilyspiniolas commented on Source/WebCore/rendering/RenderBox.cpp:
```diff
@@ -350,7 +350,7 @@ void RenderBox::styleDidChange(StyleDifference diff, const RenderStyle* oldStyle
     // new zoomed coordinate space.
     if (hasNonVisibleOverflow() && layer() && oldStyle && oldStyle->usedZoom() != newStyle.usedZoom()) {
-        if (auto* scrollableArea = layer()->scrollableArea()) {
+        if (CheckedPtr<RenderLayerScrollableArea> scrollableArea = layer()->scrollableArea()) {
```
```suggestion
        if (CheckedPtr scrollableArea = layer()->scrollableArea()) {
```
(same goes for all other uses of CheckedPtr in this PR)

kate-k-lee replied:
Thanks for the suggestion. I’ve updated the code to use CTAD (CheckedPtr scrollableArea = …).
This keeps the declaration concise and lets the pointee type (including constness) be deduced from the initializer.

lilyspiniolas replied:
Perfect!

kate-k-lee replied:
@lilyspiniolas 
Thank you for your time reviewing this patch.  
From what I can tell, the Windows EWS failure doesn’t seem related to the changes in this patch, and may be due to an infrastructure issue.  
Would you be able to re-trigger the Windows EWS, or proceed with landing if appropriate?  
I’d greatly appreciate your assistance.

lilyspiniolas replied:
Of course, thank you for your contribution! The Windows EWS failure seems unrelated to your change, so I will go ahead and land this.

kate-k-lee replied:
@lilyspiniolas  Thank you so much for taking the time to review my PR and helping me get it landed! This is my first contribution to WebKit and your guidance made it possible. I really appreciate your patience and detailed feedback throughout the process. Looking forward to contributing more! 🙏

Commit:
Use CheckedPtr for complex lifetime objects in RenderBox.cpp
https://bugs.webkit.org/show_bug.cgi?id=297189

Reviewed by Lily Spiniolas.

This patch updates auto* declarations for objects with complex lifetimes (such as ScrollableArea) to CheckedPtr to improve debugging safety while maintaining rendering performance.

Raw pointers are kept unchanged for DOM/tree navigation results where lifetime is well-defined and performance is critical.

No functional behavior changes are introduced.

Canonical link: https://commits.webkit.org/298943@main
